src/cgi.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2023 Gerhard Schiller (gerhard.schiller@protonmail.com)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * https://www.gnu.org/licenses/gpl-3.0.txt
+ *
+ * This code is based on parts of the file: "LwIP/LwIP_HTTP_Server_Raw/Src/httpd_cg_ssi.c"
+ *
+ *
+ * For details see the copyright notice below.
+ *
+ * Note: this STMicroelectronics software component has no LICENSE file
+ * in its root directory.
+ *
+ */
+
+
+/**
+  ******************************************************************************
+  * @file    LwIP/LwIP_HTTP_Server_Raw/Src/httpd_cg_ssi.c
+  * @author  MCD Application Team
+  * @brief   Webserver SSI and CGI handlers
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2017 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
```

Comment:
I'm not really in favor of adding the file with a specific author name and a different license.  What if me or someone else wants to modify/extend it later? It makes maintenance far more complex. There is a common LICENSE file in the root of this repository (it's MIT though).

If you're worried about the STMicroelectronics copyright, why not derive the example implementation from the lwIP example like the other examples in this repository?
https://github.com/lwip-tcpip/lwip/blob/master/contrib/examples/httpd/cgi_example